osbenchmark/worker_coordinator/runner.py

```diff
@@ -967,6 +968,97 @@ async def _scroll_query(opensearch, params):
                 "took": took
             }
 
+        async def _vector_search_query_with_recall(opensearch, params):
+            """
+            Perform vector search and report recall@k , recall@r and time taken to perform recall in ms as
+            meta object.
+            """
+            result = {
+                "weight": 1,
+                "unit": "ops",
+                "success": True,
+                "recall@k": 0,
+                "recall@1": 0,
+            }
+
+            def _is_empty_search_results(content):
+                if content is None:
+                    return True
+                if "hits" not in content:
+                    return True
+                if "hits" not in content["hits"]:
+                    return True
+                if len(content['hits']['hits']) == 0:
+                    return True
+                return False
+
+            def calculate_recall(predictions, neighbors, top_k):
+                """
+                Calculates the recall by comparing top_k neighbors with predictions.
+                recall = Sum of matched neighbors from predictions / total number of neighbors from ground truth
+                Args:
+                    predictions: list containing ids of results returned by OpenSearch.
+                    neighbors: list containing ids of the actual neighbors
+                    neighbors for a set of queries
```

Comment:
Nit: Should this just be `list containing ids of the actual neighbors for a set of queries` is this part of the sentence above?

Reply:
Ack